src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -148,37 +157,54 @@ public String produceDts(Compiler compiler) {
           isDefault = true;
         }
       }
-      emitNoSpace("declare namespace ");
-      emitNoSpace(INTERNAL_NAMESPACE);
-      if (!namespace.isEmpty()) {
-        emitNoSpace("." + namespace);
-      }
-      emitNoSpace(" {");
-      indent();
-      emitBreak();
-      if (symbol.getType() != null) {
-        walkScope(symbol, namespace, isDefault);
-      } else {
-        // JSCompiler treats "foo.x" as one variable name, so collect all provides that start with
-        // $provide + ".".
-        String prefix = provide + ".";
-        for (TypedVar other : topScope.getAllSymbols()) {
-          String otherName = other.getName();
-          if (otherName.startsWith(prefix) && other.getType() != null
-              && !other.getType().isFunctionPrototypeType()) {
-            walkScope(other, namespace, false);
-          }
-        }
-      }
-      unindent();
-      emit("}");
-      emitBreak();
+      currentNamespace = namespace;
```

Comment:
This IMHO introduces a rather surprising piece of global state into the translation. What about extracting a `relativeTo(namespace)` method that returns the appropriate `Function`?